src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -56,7 +59,24 @@ public FirehoseDb() {
     public void bootstrapApi() throws Exception {
         List<Pair<String, String>> nameActuals = new ArrayList<>();
         // our local streams
-        getFirehoseClient().listDeliveryStreams(new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut")).getDeliveryStreamNames().forEach(name -> nameActuals.add(Pair.of(name.toLowerCase(), name)));
+        // getFirehoseClient().listDeliveryStreams(new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut")).getDeliveryStreamNames().forEach(name -> nameActuals.add(Pair.of(name.toLowerCase(), name)));
+
+        ListDeliveryStreamsRequest listDeliveryStreamsRequest = new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut");
+        listDeliveryStreamsRequest.setLimit(listDeliveryStreamsLimit);
+        ListDeliveryStreamsResult listDeliveryStreamsResult = getFirehoseClient().listDeliveryStreams(listDeliveryStreamsRequest);
+        List<String> deliveryStreamNames = listDeliveryStreamsResult.getDeliveryStreamNames();
+
+        while (listDeliveryStreamsResult.getHasMoreDeliveryStreams())
+        {
+            if (!deliveryStreamNames.isEmpty()) {
+                listDeliveryStreamsRequest.setExclusiveStartDeliveryStreamName(deliveryStreamNames.get(deliveryStreamNames.size() - 1));
```

Comment:
This is really verbose - and as you noted by doing the if statement, not safe because if its empty there is no -1.  Also, you used a magic number (1).  sus.

Reply:
Agreed, updating to use `reduce()` as mentioned below.